Add .pc files for the OpenGLESv2, EGL and bcm_host libraries

Those pkg-config files make it easier for Qt5 to find those libraries
and the appropriate link flags.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

Index: b/interface/khronos/egl/egl.pc.in
===================================================================
--- /dev/null
+++ b/interface/khronos/egl/egl.pc.in
@@ -0,0 +1,13 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: egl
+Description: RasberryPi implementation of EGL
+Version: 10
+Libs: -L${libdir} -lEGL -lGLESv2 -lbcm_host
+Cflags: -I${includedir}/ \
+        -I${includedir}/interface/vcos/pthreads/ \
+        -I${includedir}/interface/vmcs_host/ \
+        -I${includedir}/interface/vmcs_host/linux/
Index: b/interface/khronos/CMakeLists.txt
===================================================================
--- a/interface/khronos/CMakeLists.txt
+++ b/interface/khronos/CMakeLists.txt
@@ -74,3 +74,11 @@
 
 install(TARGETS EGL GLESv2 OpenVG WFC khrn_client DESTINATION lib)
 install(TARGETS EGL_static GLESv2_static khrn_static DESTINATION lib)
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/egl/egl.pc.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/egl/egl.pc" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/egl/egl.pc"
+  DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/glxx/glesv2.pc.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/glxx/glesv2.pc" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/glxx/glesv2.pc"
+  DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
Index: b/interface/khronos/glxx/glesv2.pc.in
===================================================================
--- /dev/null
+++ b/interface/khronos/glxx/glesv2.pc.in
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: glesv2
+Description: RasberryPi implementation of OpenGL ESv2
+Version: 2.0
+Libs: -L${libdir} -lGLESv2
+Cflags: -I${includedir}/
Index: b/host_applications/linux/libs/bcm_host/CMakeLists.txt
===================================================================
--- a/host_applications/linux/libs/bcm_host/CMakeLists.txt
+++ b/host_applications/linux/libs/bcm_host/CMakeLists.txt
@@ -20,3 +20,7 @@
 
 install(TARGETS bcm_host DESTINATION lib)
 
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/bcm_host.pc.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/bcm_host.pc" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/bcm_host.pc"
+  DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
Index: b/host_applications/linux/libs/bcm_host/bcm_host.pc.in
===================================================================
--- /dev/null
+++ b/host_applications/linux/libs/bcm_host/bcm_host.pc.in
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: bcm_host
+Description: Broadcom VideoCore host API library
+Version: 1
+Libs: -L${libdir} -lbcm_host -lvcos -lvchiq_arm
+Cflags: -I${includedir} -I${includedir}/interface/vmcs_host/linux -I${includedir}/interface/vcos/pthreads -DUSE_VCHIQ_ARM
